source/find-global-type.d.ts

```diff
@@ -0,0 +1,48 @@
+/**
+Tries to find one or more types from their globally-defined constructors.
+
+Use-case: Conditionally referencing DOM types only when the DOM library is present.
+
+Note: Globally-defined has a narrow definition in this case due to peculiarities with the behavior of `globalThis`. Merely declaring a class in a `declare global` block won't work, instead you must declare its constructor as a `var` (not `let`/`const`) inside the `declare global` block. This typically is done using a combination of interface + constructor style, rather than class syntax.
+
+@example
+```
+import type {FindGlobalType} from 'type-fest';
+
+class Point {
+	constructor(public x: number, public y: number) {}
+}
+
+type PointLike = Point | FindGlobalType<'DOMPoint'>;
+
+function doSomething(point: PointLike): number {
+	// …
+}
+```
+
+@example
+```
+import type {FindGlobalType} from 'type-fest';
+
+declare global {
+	// Class syntax won't add the key to `globalThis`
+	class Foo {}
+
+	// Interface + constructor style works
+	interface Bar {}
+	interface BarConstructor {
+		new (): Bar;
+	}
+	var Bar: BarConstructor; // Not let or const
+}
+
+type FindFoo = FindGlobalType<'Foo'>; // Doesn't work
+type FindBar = FindGlobalType<'Bar'>; // Works
+```
+
+@category Utilities
+*/
+export type FindGlobalType<Name extends string> =
```

Comment:
Should we name this `FindGlobalInstanceType` or something instead? As it only looks for global constructors, if used with eg `FindGlobalType<'fetch'>` it won't give the type?

Reply:
[`InstanceType`](https://www.typescriptlang.org/docs/handbook/utility-types.html#instancetypetype) being the name of the built in utility type for finding the instance type of a constructor

Reply:
Maybe something like this?

```ts
type FindGlobalType<Name extends string> =
  typeof globalThis extends Record<Name, infer T>
    ? T
    : never;

type FindGlobalInstanceType<Name extends string> =
  FindGlobalType<Name> extends new (...args: any[]) => infer T
    ? T
    : never;
```

Reply:
That name would probably be better, I wasn't aware of that builtin, and splitting it up like that would actually be nice.

If we wanted to be a bit more precise, we could also use `FindInstanceTypeFromGlobal` or `FindTypeFromGlobalConstructor`. I would love to have a more versatile type for this, since it won't even work for a `declare global { class Foo { } }`, but I don't know any way to do that.

Reply:
Just pushed a rename and split to `FindGlobalType`/`FindGlobalInstanceType`. For reasons I wasn't able to determine, `FindGlobalInstanceType` would return `unknown` for non-existent keys when delegating to `FindGlobalType`, so it still directly references `globalThis`